lib/src/ruby/api.cc

```diff
@@ -485,21 +486,47 @@ namespace facter { namespace ruby {
         }
         LOG_DEBUG("ruby was found at \"%1%\".", ruby);
 
-        bool success;
-        string output, none;
-        tie(success, output, none) = execute(ruby, { "-e", "print File.join(RbConfig::CONFIG['"+libruby_configdir()+"'], RbConfig::CONFIG['LIBRUBY_SO'])" });
+        bool success, loaded = false;
+        string ruby_cmd_args, ruby_output, libruby_file, libruby_full_path, none;
+        tie(success, libruby_file, none) = execute(ruby, { "-e", "print RbConfig::CONFIG['LIBRUBY_SO']"});
```

Comment:
Or don't worry about the formatting and just always use `['libdir', 'archlibdir', 'sitearchlibdir'. 'bindir']`?